hls.go

```diff
@@ -0,0 +1,765 @@
+			streams, err := cursor.Streams()
+			if err == nil && streams != nil && len(streams) > 0 {
+				// Write stream headers to the TS muxer
+				err := newMuxer.WriteHeader(streams)
+				if err != nil {
+					common.LogErrorf("Failed to write stream headers to TS muxer: %v\n", err)
+					// Fall back to creating muxer without headers, but this may cause issues
+				} else {
+					common.LogDebugf("TS muxer initialized with %d streams\n", len(streams))
+				}
+			} else {
+				common.LogDebugf("No streams available for TS muxer initialization\n")
+			}
+		}
+	}
+
+	*muxer = newMuxer
+
+	common.LogDebugf("Started new HLS segment\n")
+}
+
+// finalizeSegment completes the current segment and adds it to the playlist
+func (h *HLSChannel) finalizeSegment(buffer *bytes.Buffer, duration time.Duration) {
+	if buffer.Len() == 0 {
+		return
+	}
+
+	// Create a copy of the buffer data
+	segmentData := make([]byte, buffer.Len())
+	copy(segmentData, buffer.Bytes())
+
+	currentSeq := h.sequenceNumber
+	h.sequenceNumber++
+
+	// Generate unique segment ID to avoid browser caching issues across service restarts
+	segmentID := generateSegmentID()
+	segmentURI := fmt.Sprintf("/live/segment_%s.ts", segmentID)
+	durationSeconds := duration.Seconds()
+
+	segment := HLSSegment{
+		URI:      segmentURI,
+		Duration: durationSeconds,
+		Data:     segmentData,
+		Sequence: currentSeq, // Keep sequence for internal ordering
+	}
+
+	// Add segment with proper sliding window management
+	h.addGeneratedSegment(segment)
+
+	common.LogDebugf("Finalized HLS segment %d with %d bytes, duration %.2fs\n",
+		currentSeq, len(segmentData), durationSeconds)
+}
+
+// createSegment creates a new HLS segment
+// addGeneratedSegment adds a generated segment to the HLS channel with proper sliding window
+func (h *HLSChannel) addGeneratedSegment(segment HLSSegment) {
+	if h == nil || h.playlist == nil {
+		return
+	}
+
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+
+	// Add segment to our local list with sliding window management
+	h.segments = append(h.segments, segment)
+
+	// Remove old segments if we exceed max (manual sliding window for our data)
+	if len(h.segments) > h.maxSegments {
+		// Remove oldest segments to maintain window size
+		excess := len(h.segments) - h.maxSegments
+		h.segments = h.segments[excess:]
+	}
+
+	// If this is the first segment, set the initial MediaSequence
+	if h.playlist.Count() == 0 {
+		h.playlist.SeqNo = segment.Sequence
+	}
+
+	// For proper sliding window, we need to manually manage the playlist size
+	// If the playlist is at max capacity, we need to remove the oldest segment first
+	if int(h.playlist.Count()) >= h.maxSegments {
+		// Create a new playlist and copy the recent segments
+		newPlaylist, err := m3u8.NewMediaPlaylist(uint(h.maxSegments), uint(h.maxSegments))
+		if err != nil {
+			common.LogErrorf("Failed to create new playlist for sliding window: %v\n", err)
+			return
+		}
+		newPlaylist.SetVersion(6)
```

Comment:
At this point, how often we need to set the version to 6 it MAY be an idea to put the version into a variable in case it needs to be changed at a later point